src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -273,6 +273,8 @@ private Activity StartActivity(HttpContext httpContext)
                 }
             }
 
+            _diagnosticListener.OnActivityExport(activity, httpContext.Request);
```

Comment:
Should this be the HttpRequest or some anonymous objects?

Reply:
This is intended to be request. 

Request is passed into OnActivityExport, OnActivityImport and IsEnabled callbacks. They should be used to quickly filter activities. Write events get anonymous objects.

Reply:
If you ever need anything else, it’ll be a breaking change (or I guess we can shuttle it through the HttpContext)

Reply:
The main scenario for the callback is setting trace flags (sampling bit). All info to make the decision should be already set on Activity (trace context). 

Even request is optional. There is not much info available at this point except the Request and it's hard to imagine a legit scenario where we'll need more.

Reply:
Ok, I changed the extra argument in OnActivityImport to be HttpContext (not request) just in case we'll ever need to extend it